### ISSUE TITLE
[Discussion] Library goal and what should be here

### ISSUE THREAD

**Issue by drdanz** [robotology-playground/common-classes#3]
I added a lot of stuff that I would like to have here in the main `README.md` file, but I think we should discuss point by point

**Comment by aerydna** [comment]
this library is intended to serve 2 main necessity,

- a place where we can put general utility, classes and methods not present in the standard c++ but useful for robotology, also copying from other famous c++ library
- a sort of long buffer for using c++ advanced feature. offering similar functionality, then Wrapping the standard version whenever it come out and then slowly deprecating them asking downstream to simply substitute the `robotology::xxx` with `std::yyy`.

**Comment by traversaro** [comment]
A sort of robotology's [Abseil](https://abseil.io/) (but hopefully with a release model more aligned with standard open source practice)?

**Comment by aerydna** [comment]
exactly, with also utility more related to the robotology world if we want
